GUI/window.py

```diff
@@ -506,26 +505,42 @@ def save_io_ports(self, *args):
             title = args[1].title
             text = args[1].text_field.text
             if text.isdigit():
-                num = int(text)
-                if num < 0 or num > 4095:
+                port = int(text)
+                if port < 0 or port > 4095:
                     toast('Invalid port number. Valid port numbers [0-4095]')
                 else:
-                    toast_message = 'Changed port number'
-                    if title == TRAFFIC_LIGHT['menu_title']:
-                        TRAFFIC_LIGHT['port'] = num
-                    elif title == SEVEN_SEGMENT_DISPLAY['menu_title']:
-                        SEVEN_SEGMENT_DISPLAY['port'] = num
-                    elif title == ASCII_TABLE['menu_title']:
-                        if num > 4088:
-                            toast_message = 'Invalid port for ASCII Table. Valid ports [0-4088]'
+                    if is_valid_port(port):
+                        if title == TRAFFIC_LIGHT['menu_title']:
+                            update_reserved_ports(TRAFFIC_LIGHT['port'], port)
+                            TRAFFIC_LIGHT['port'] = port
```

Comment:
`TRAFFIC_LIGHT['port'] = port` <--- Do this inside of `update_reserved_ports`

And do the same for the rest of the I/O

Reply:
What do you mean? `update_reserved_ports` is used to update the stack using the old value of that I/O, not update the dictionary where `TRAFFIC_LIGHT` is located.

Reply:
AH. My bad, read wrong. Read do this **instead of**, not **inside of**.